### PR TITLE
Adds support for HSSD (Habitat Synthetic Scene Dataset)

### PR DESCRIPTION
**Pull request by mukulkhanna**
## Motivation and Context

This PR adds dataset and training config files, dataset download links, etc. for the new HSSD dataset.

## How Has This Been Tested

Added tests for HSSD-hab and ProcTHOR-hab for ObjecNav dataset and task setup and baseline training.

```
pytest -v test/test_baseline_training.py
pytest -v test/test_object_nav_task.py
```

Note that these tests will not run on CircleCI because it wouldn't make sense to download the entire HSSD and AI2THOR datasets for each test. These tests passed locally.

## Types of changes

- **\[Docs change\]** Addition or changes to the documentation
- **\[Refactoring\]** Large changes to the code that improve its functionality or performance
- **\[Dependency Upgrade\]** Upgrades one or several dependencies in habitat
- **\[Bug Fix\]** (non-breaking change which fixes an issue)
- **\[Development\]** A pull request that add new features to the [habitat-lab](/habitat-lab) task and environment codebase. Development Pull Requests must be small (less that 500 lines of code change), have unit testing, very extensive documentation and examples. These are typically new tasks, environments, sensors, etc... The review process for these Pull Request is longer because these changes will be maintained by our core team of developers, so make sure your changes are easy to understand!
- **\[Experiment\]** A pull request that add new features to the [habitat-baselines](/habitat-baselines/) training codebase. Experiments Pull Requests can be any size, must have smoke/integration tests and be isolated from the rest of the code. Your code additions should not rely on other habitat-baselines code. This is to avoid dependencies between different parts of habitat-baselines. You must also include a README that will document how to use your new feature or trainer. **You** will be the maintainer of this code, if the code becomes stale or is not supported often enough, we will eventually remove it.

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

File: DATASETS.md
Comment (vincentpierre) [inline DATASETS.md:1]: Maybe remove the old 🆕 emojis and put the new datasets at the bottom ?

File: habitat-baselines/habitat_baselines/config/objectnav/ddppo_objectnav_hssd.yaml
Comment (vincentpierre) [inline habitat-baselines/habitat_baselines/config/objectnav/ddppo_objectnav_hssd.yaml:1]: Have all of these new configurations been tested and if yes, can you add unit tests?

**Review by vincentpierre — APPROVED** [review]
Please add unit tests before merging.

File: habitat-lab/habitat/config/habitat/dataset/objectnav/hssd.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# @package habitat.dataset
+defaults:
+  - /habitat/dataset: dataset_config_schema
+  - _self_
+
+type: ObjectNav-v1
+split: train
+data_path: data/datasets/objectnav/hssd/v0.2.3/{split}/{split}.json.gz
```
[the remark this answers is not in the document]
Comment (home-robot-ovmm) [reply habitat-lab/habitat/config/habitat/dataset/objectnav/hssd.yaml:8]: We discussed this and felt it was easier to maintain the same numbering for the internal and external (public-facing) releases.

CC: @aclegg3 @msavva

[the remark this answers is not in the document]
Comment (mukulkhanna) [reply habitat-lab/habitat/config/habitat/dataset/objectnav/hssd.yaml:8]: We discussed this and felt it was easier to maintain the same numbering for the internal and external (public-facing) releases.

CC: @aclegg3 @msavva

File: habitat-baselines/habitat_baselines/config/objectnav/ddppo_objectnav_hssd.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# @package _global_
+
+defaults:
+  - /habitat_baselines: habitat_baselines_rl_config_base
+  - /benchmark/nav/objectnav: objectnav_hssd
+  - /habitat_baselines/rl/policy/obs_transforms:
+    - center_cropper_base
```
[the remark this answers is not in the document]
Comment (mukulkhanna) [reply habitat-baselines/habitat_baselines/config/objectnav/ddppo_objectnav_hssd.yaml]: Fixed, thanks!

**Comment by mukulkhanna** [comment]
@aclegg3 @vincentpierre 

I have added unit tests for HSSD-hab and ProcTHOR-hab for ObjecNav dataset and task setup along with baseline training.

```
pytest -v test/test_baseline_training.py
pytest -v test/test_object_nav_task.py
```

Note that these tests will be skipped on CircleCI because (as @aclegg3 suggested) it wouldn't make sense to download the entire HSSD and AI2THOR datasets for each test. These tests passed locally for me.

**Review by aclegg3 — APPROVED** [review]
Overall logic and docs look good to me. Others should comment on the config files as necessary.

**Pull request merged**